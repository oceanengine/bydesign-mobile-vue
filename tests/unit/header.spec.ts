import { mount } from '../index';
import { Header as NativeHeader } from '@src'
const Header = NativeHeader as any;
describe('Header', () => {
  test('should render left/right slot and match snapshot', () => {
    const wrapper = mount(Header, {
      propsData: {
        title: 'Title',
      },
      scopedSlots: {
        left: () => 'Custom Left',
        right: () => 'Custom Right',
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
  test('should render title slot and match snapshot', () => {
    const wrapper = mount(Header, {
      scopedSlots: {
        title: () => 'Custom Title',
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  test('should emit click-left event when clicking left text', () => {
    const wrapper = mount(Header, {
      propsData: {
        leftArrow: true,
      },
    });

    wrapper.find('.byted-header__left').trigger('click');
    expect(wrapper.emitted('click-left')).toBeTruthy();
  });
  test('should emit click-right event when clicking right text', () => {
    const wrapper = mount(Header, {
      scopedSlots: {
        left: () => 'Custom Left',
        right: () => 'Custom Right',
      },
    });

    wrapper.find('.byted-header__right').trigger('click');
    wrapper.find('.byted-header__left').trigger('click');
    expect(wrapper.emitted('click-right')).toBeTruthy();
    expect(wrapper.emitted('click-left')).toBeTruthy();
  });
});
