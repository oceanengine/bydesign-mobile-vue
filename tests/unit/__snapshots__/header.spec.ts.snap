// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render left/right slot and match snapshot 1`] = `
<div class="byted-header byted-header--default byted-hairline--bottom">
  <div class="byted-header__left">Custom Left</div>
  <div class="byted-header__title">Title</div>
  <div class="byted-header__right">Custom Right</div>
</div>
`;

exports[`Header should render title slot and match snapshot 1`] = `
<div class="byted-header byted-header--default byted-hairline--bottom">
  <div class="byted-header__left"><i class="byted-header__icon" style="width: 1.2rem; height: 1.2rem;">test-file-stub</i></div>
</div>
`;
